File: src/day3.py
import utils

def __main__():
    lines = utils.read_input()

    result = 0

    x = 0
    for line in lines[1:]:
        x = (x + 3) % len(line)
        if line[x] == '#':
            result += 1

    print(result)

__main__()

File: rust/src/day3.rs
mod utils;

fn main() {
    let lines = utils::read_input();

    let mut result = 0;
    let mut x = 0;

    lines.iter().for_each(|line| {
        if line.as_bytes()[x] == b'#' {
            result += 1;
        }

        x = (x + 3) % line.len();
    });

    println!("{}", result);
}
